### ISSUE TITLE
`assert!` on `==` should be `assert_eq!`

### ISSUE THREAD

**Issue by Apanatshka** [rust-lang/rust-clippy#12112]
### What it does

1. Find uses of `assert!` on the result of [std::cmp::PartialEq::eq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html#tymethod.eq), suggest using `assert_eq!`.
2. Do the same for `debug_assert_eq!`.

### Advantage

An assertion failure can output a more useful message.

### Example

```rust
assert!(
    self.some_method()
        == local_value
            .get_other_information(self.complex.expression[0], this.is.long),
    "The some_method result must be in sync with the other information."
);
```

Could be written as:

```rust
assert_eq!(
    self.some_method(),
    local_value.get_other_information(self.complex.expression[0], this.is.long),
    "The some_method result must be in sync with the other information."
);
```

**Comment by nicklimmm** [comment]
The lint for the above is deprecated from #2156.

From https://github.com/rust-lang/rust/issues/44838, seems like we are heading towards a better `assert!` macro with better messages.

**Comment by Apanatshka** [comment]
I was unaware of the planned changes, happy to hear that things will becomes better automatically :)
